vault/identity_store_util.go

```diff
@@ -2035,14 +2035,6 @@ func (i *IdentityStore) UpsertGroupInTxn(ctx context.Context, txn *memdb.Txn, gr
 		return fmt.Errorf("group is nil")
 	}
 
-	g, err := i.MemDBGroupByName(ctx, group.Name, true)
```

Comment:
🤔 I wonder if this would have worked if we used the txn that's passed into this method...

Reply:
It would not have. The problem here is the lines below where we set the incoming group.ID to be the same as the existing one. That meant that when there are duplicate groups in storage we end up loading the second one over the first, but switching it to have the first one's ID so now you can't even lookup the original by ID and the one you do get is a strange hybrid of two different groups in storage...